ReactiveCocoa/Swift/Property.swift

```diff
@@ -240,12 +240,18 @@ public func <~ <P: MutablePropertyType>(property: P, signal: Signal<P.Value, NoE
 public func <~ <P: MutablePropertyType>(property: P, producer: SignalProducer<P.Value, NoError>) -> Disposable {
 	let disposable = CompositeDisposable()
 
-	producer.startWithSignal { signal, signalDisposable in
-		property <~ signal
-		disposable += signalDisposable
-
-		disposable += property.producer.startWithCompleted {
+	disposable += property.producer.startWithCompleted {
+		disposable.dispose()
+	}
+	
+	disposable += producer.start { [weak property] event in
+		switch event {
+		case let .Next(value):
+			property?.value = value
```

Comment:
Yeah :( But attempt to reuse canonical one does not look good either and there already was a PR that solved one leak. I think in this case it'd be better to have duplication of simple solution.
If you think otherwise I can revert to initial PR.